### DISCUSSION TITLE
Dynamic login ui url

### DISCUSSION THREAD

**Discussion by robbietree8** [ory/kratos#994] [Q&A] [answered]
We have multiple applications that needs authentication, like app1 and app2.
There will be cases where different UI look and feel is needed. For example, app1 is hosted in www.example.com/app1, and the login page will be www.example.com/app1/login, app2 is hosted in www.example.com/app2, and the login page will be www.example.com/app2/login.
Currently from the doc, only one login ui url is allowed.

Is it possible to provide login endpoint when initializing login flow for browsers?

**Comment by zepatrik** [comment, marked as answer]
There was a similar topic in the slack: https://ory-community.slack.com/archives/C012RJ2MQ1H/p1608063605048300

> ZipperSK  9:20 PM
> Hi! We are using Kratos for website that is hosted on multiple top level domains. [...]
>
> ZipperSK  9:35 PM
> The problem is taht I need to specify multiple ui_url . One for each tld. So if I initialize registration flow from mydomain.sk a would be redirected with flowId to mydomian.sk and if it is from mydomain.cz I would be redirected to mydomain.cz
> I could maybe make multiple Kratos instances (one for each subdomain), but I wonder if there is any simpler solution 
> 
> ZipperSK  9:51 PM
> So I studied a code for a bit (this part https://github.com/ory/kratos/blob/5f88465a9edb0f00a8b26fa4518abf5fb4c4ef76/selfservice/flow/registration/handler.go#L152 ), and it does not seem possible to specify redirect_to  or mutliple ui_url . So I will probably run multiple Kratos instances with different UI urls in configs.
>
> Patrik (ory)  9:21 AM
> Different UI URLs are not possible [...]
> Consider running the kratos UIs under a single domain and redirect people back to where they came from
This would then be the same as e.g. Google, where you log in under google.com while you actually use youtube
>
> Patrik (ory) 9:22
These multi domain setups are something we are still working on, but we want to support that [edit: see #662]
>
> ZipperSK  10:57 AM
> we determine user language by top level domain (so .cz site is translated in Czech and .sk in Slovak), so running  kratos UIs under single domain would not be possible for us without some reimplementation in our system. I tried running multiple Kratos docker instances with same config, and overwritte ui url by environmnet variable specified in docker compose. Everythink seems to work. (edited) 
>
> Patrik (ory)  10:59 AM
> do you use the same dsn?
>
> ZipperSK  11:00 AM
> dsn is postgres database? :smile:
> ZipperSK 11:01
> so yes, I am using same dsn postgres database that is with kratos in internal netowrk
>
> Patrik (ory)  11:02 AM
> hm ok nice approach, I don't see why it would not work :thinking:

**Reply by robbietree8** [reply]
Thanks for your quick reply, I did search the issues/docs/discussions, but not slack, will do it next time.

**Reply by zepatrik** [reply]
No problem, it is not easy to find there. That's why I transferred the whole thread over here, so someone will be able to find it in the future.
